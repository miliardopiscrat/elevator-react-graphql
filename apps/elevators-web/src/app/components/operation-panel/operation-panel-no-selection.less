.selection-message {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  > * + * {
    margin-left: 0.5em;
  }
}

