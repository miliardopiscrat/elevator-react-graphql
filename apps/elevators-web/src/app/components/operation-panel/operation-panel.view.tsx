import React, { FC, PropsWithChildren } from 'react';
import Card from 'antd/lib/card';

export const OperationPanelView: FC<PropsWithChildren<{}>> = ({ children }) => {
  return <Card title="Elevator operations">{children}</Card>;
};
