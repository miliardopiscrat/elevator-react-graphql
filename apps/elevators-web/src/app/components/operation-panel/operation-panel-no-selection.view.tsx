import React, { memo } from 'react';
import Icon from 'antd/lib/icon';
import { OperationPanelView } from './operation-panel.view';
import './operation-panel-no-selection.less';

export const OperationPanelNoSelection = memo(() => {

  return <OperationPanelView>
    <div className='selection-message'>
      <Icon title="Information" type='info-circle' style={{ fontSize: '2em', color: '#3b79c0' }}/>
      <span>Please select Elevator!</span>
    </div>
  </OperationPanelView>;
});
