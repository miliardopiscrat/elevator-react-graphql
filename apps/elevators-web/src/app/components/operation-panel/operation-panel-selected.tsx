import React, { FC } from 'react';
import { OperationPanelView } from './operation-panel.view';
import { OperationMoveElevatorContainer } from '../operation-move-elevator/operation-move-elevator.container';

interface PropsType {
  selectedElevatorId: number;
}

export const OperationPanelSelected: FC<PropsType> = ({selectedElevatorId}) => {

  return  <OperationPanelView>
    <OperationMoveElevatorContainer elevatorId={selectedElevatorId}/>
  </OperationPanelView>;
};
