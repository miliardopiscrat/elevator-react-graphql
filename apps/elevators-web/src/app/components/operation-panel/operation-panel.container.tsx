import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import { getSelectedElevatorGQL, SelectedElevatorIdType } from '../../core/graph-tags/get-selected-elevator.client';
import { OperationPanelNoSelection } from './operation-panel-no-selection.view';
import { OperationPanelSelected } from './operation-panel-selected';

export const OperationPanelContainer = () => {
  const { data } = useQuery<SelectedElevatorIdType>(getSelectedElevatorGQL);

  if (!data || data.selectedElevatorId === null) return <OperationPanelNoSelection/>;

  return <OperationPanelSelected selectedElevatorId={data.selectedElevatorId}/>;
};
