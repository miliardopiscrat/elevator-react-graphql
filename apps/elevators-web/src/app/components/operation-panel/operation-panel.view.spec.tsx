import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { OperationPanelView } from './operation-panel.view';

describe('OperationPanelView', () => {
  afterEach(cleanup);

  it('should render successfully', () => {
    const { baseElement } = render(<OperationPanelView/>);

    expect(baseElement).toBeTruthy();
  });

  it('should render Card with title', () => {
    const { getByText } = render(<OperationPanelView/>);

    expect(getByText('Elevator operations')).toBeInTheDocument();
  });

  it('should match content', () => {
    const { container } = render(<OperationPanelView>TEST_CONTENT</OperationPanelView>);

    expect(container.innerHTML).toMatchSnapshot();
  });
});
