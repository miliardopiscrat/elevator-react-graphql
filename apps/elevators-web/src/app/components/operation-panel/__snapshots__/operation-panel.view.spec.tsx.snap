// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperationPanelView should match content 1`] = `"<div class=\\"ant-card ant-card-bordered\\"><div class=\\"ant-card-head\\"><div class=\\"ant-card-head-wrapper\\"><div class=\\"ant-card-head-title\\">Elevator operations</div></div></div><div class=\\"ant-card-body\\">TEST_CONTENT</div></div>"`;
